packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -2039,6 +2046,854 @@ contract OPContractsManagerInteropMigrator is OPContractsManagerBase {
+    }
+
+    /// @notice Resets the initialized slot for a contract and then initializes it.
+    /// @param _proxyAdmin The proxy admin of the contract.
+    /// @param _target The target of the contract.
+    /// @param _implementation The implementation of the contract.
+    /// @param _data The data to call the initializer with.
+    /// @param _slot The slot where the initialized value is located.
+    /// @param _offset The offset of the initializer value in the slot.
+    function _resetAndInitialize(
+        IProxyAdmin _proxyAdmin,
+        address _target,
+        address _implementation,
+        bytes memory _data,
+        bytes32 _slot,
+        uint8 _offset
+    )
+        internal
+    {
+        // Upgrade to StorageSetter.
+        _proxyAdmin.upgrade(payable(_target), address(implementations().storageSetterImpl));
+
+        // Reset the initialized slot by zeroing the single byte at `_offset` (from the right).
+        bytes32 current = StorageSetter(_target).getBytes32(_slot);
+        uint256 mask = ~(uint256(0xff) << (uint256(_offset) * 8));
+        StorageSetter(_target).setBytes32(_slot, bytes32(uint256(current) & mask));
+
+        // Upgrade to the implementation and call the initializer.
+        _proxyAdmin.upgradeAndCall(payable(address(_target)), _implementation, _data);
+    }
+
+    /// @notice Helper for making the SystemConfig initializer arguments. This is the only
+    ///         initializer that needs a helper function because we get stack-too-deep.
+    /// @param _cfg The full config.
+    /// @param _cts The chain contracts.
+    /// @return The SystemConfig initializer arguments.
+    function _makeSystemConfigInitArgs(
+        FullConfig memory _cfg,
+        ChainContracts memory _cts
+    )
+        internal
+        pure
+        returns (bytes memory)
+    {
+        // Generate the SystemConfig addresses input.
+        ISystemConfig.Addresses memory addrs = ISystemConfig.Addresses({
+            l1CrossDomainMessenger: address(_cts.l1CrossDomainMessenger),
+            l1ERC721Bridge: address(_cts.l1ERC721Bridge),
+            l1StandardBridge: address(_cts.l1StandardBridge),
+            optimismPortal: address(_cts.optimismPortal),
+            optimismMintableERC20Factory: address(_cts.optimismMintableERC20Factory),
+            delayedWETH: address(_cts.delayedWETH)
+        });
+
+        // Generate the initializer arguments.
+        return abi.encodeCall(
+            ISystemConfig.initialize,
+            (
+                _cfg.systemConfigOwner,
+                _cfg.basefeeScalar,
+                _cfg.blobBasefeeScalar,
+                bytes32(uint256(uint160(_cfg.batcher))),
+                _cfg.gasLimit,
+                _cfg.unsafeBlockSigner,
+                _cfg.resourceConfig,
+                chainIdToBatchInboxAddress(_cfg.l2ChainId),
+                addrs,
+                _cfg.l2ChainId,
+                _cfg.superchainConfig
+            )
```

Comment:
Alternatively, let's refactor this initializer to use some structs, then we won't get the stack too deep problem here and it will be easier to extend this in the future

Reply:
Might be a good future improvement, I didn't want to modify too much in the contracts for now